20230308/mike.java

```diff
@@ -0,0 +1,60 @@
+/*
+Use Binary Search to solve it, just same as recent daily questions.
+left can be 0 or 1, right could be max(piles) because 1 hour can only eat 1 stack.
+
+Edge case need to be careful:
+1. piles's length = 1 or 2
+*/
+class Solution {
+    int[] mPiles;
+    int mH;
+
+    public int minEatingSpeed(int[] piles, int h) {
+        Arrays.sort(piles);
```

Comment:
is sorting required?

Reply:
Because I write the wrong code at the beginning.
I forget that left is possible to be 1. That's why I sort it and choose first index as my left.
But it still work if I replace it to min() or max().
Great suggestion!